editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,629 @@
+    responses(
+        (status = 201, body = ScenarioResponseV2, description = "The created scenario"),
+    )
+)]
+#[post("")]
+async fn create(
+    db_pool: Data<DbPool>,
+    data: Json<ScenarioCreateForm>,
+    path: Path<(i64, i64)>,
+) -> Result<Json<ScenarioResponseV2>> {
+    let conn = &mut db_pool.get().await?;
+    let (project_id, study_id) = path.into_inner();
+    let timetable_id = data.timetable_id;
+    let scenario: Changeset<ScenarioModel> = data.into_inner().into();
+
+    // Check if the project and the study exist
+    let (mut project, study) = check_project_study_conn(conn, project_id, study_id).await?;
+
+    // Check if the timetable exists
+    let _ = Timetable::retrieve_or_fail(conn, timetable_id, || ScenarioError::TimetableNotFound {
+        timetable_id,
+    })
+    .await?;
+
+    // Create Scenario
+    let scenario = scenario.study_id(study_id).create(conn).await?;
+
+    // Update study last_modification field
+    study.clone().update_last_modified(conn).await?;
+
+    // Update project last_modification field
+    project.update_last_modified(conn).await?;
+
+    let scenarios_with_details = scenario.with_details_conn(conn).await?;
+
+    let scenarios_response = ScenarioResponseV2::new(scenarios_with_details, project, study);
+    Ok(Json(scenarios_response))
+}
+
+/// Delete a scenario
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    responses(
+        (status = 204, description = "The scenario was deleted successfully"),
+        (status = 404, body = InternalError, description = "The requested scenario was not found"),
+    )
+)]
+#[delete("")]
+async fn delete(path: Path<(i64, i64, i64)>, db_pool: Data<DbPool>) -> Result<HttpResponse> {
+    use crate::modelsv2::DeleteStatic;
+    let (project_id, study_id, scenario_id) = path.into_inner();
+
+    // Check if the project and the study exist
+    let (mut project, study) = check_project_study(db_pool.clone(), project_id, study_id)
+        .await
+        .unwrap();
+
+    let conn = &mut db_pool.get().await?;
+
+    // Delete scenario
+    ScenarioModel::delete_static_or_fail(conn, scenario_id, || ScenarioError::NotFound {
+        scenario_id,
+    })
+    .await?;
+
+    // Update project last_modification field
+    project.update_last_modified(conn).await?;
+
+    // Update study last_modification field
+    study.clone().update_last_modified(conn).await?;
+
+    Ok(HttpResponse::NoContent().finish())
+}
+
+/// This structure is used by the patch endpoint to patch a study
+#[derive(Serialize, Deserialize, Derivative, ToSchema)]
+#[derivative(Default)]
+struct ScenarioPatchFormV2 {
+    pub name: Option<String>,
+    pub description: Option<String>,
+    pub tags: Option<Vec<Option<String>>>,
+    pub infra_id: Option<i64>,
+}
+
+impl From<ScenarioPatchFormV2> for <ScenarioModel as crate::modelsv2::Model>::Changeset {
+    fn from(scenario: ScenarioPatchFormV2) -> Self {
+        Self {
```

Comment:
For each changeset builder method `.attr(x: X)`, you have a `.flat_attr(x: Option<X>)` to handle this precise use case. This way you may avoid having public cs fields.